### PR TITLE
Improve HTTP::Server docs

### PR DESCRIPTION
**Pull request by straight-shoota**
This is a general overhaul of the `HTTP::Server` API docs, describing the updated behaviour after #5776.
It adds some more details and structures the text.

I've left out the section about reusing connections which will be added with the changed behaviour in #7055.

### CODE REVIEW

File: src/http/server.cr [hunk 1]
```diff
@@ -54,14 +51,45 @@ require "./common"
 #   context.response.print "Hello world!"
 # end
 #
-# server.bind_tcp "0.0.0.0", 8080
-# puts "Listening on http://0.0.0.0:8080"
+# address = server.bind_tcp "0.0.0.0", 8080
+# puts "Listening on http:/#{address}"
```
Comment (bew) [inline src/http/server.cr]: typo: missing a `/`

File: src/http/server.cr [hunk 2]
```diff
@@ -39,12 +21,27 @@ require "./common"
...
-# puts "Listening on http://127.0.0.1:8080"
+# address = server.bind_tcp 8080
+# puts "Listening on http://#{address}"
 # server.listen
 # ```
 #
-# ### With non-localhost bind address
+# ## Binding to sockets
+#
+# The server can be bound to one ore more server sockets (see )
```
Comment (r00ster91) [inline src/http/server.cr]:
```suggestion
# The server can be bound to one ore more server sockets (see `#bind`)
```

File: src/http/server.cr [hunk 3]
```diff
@@ -39,12 +21,27 @@ require "./common"
...
+#
+# Supported types:
+#
+# * TCP socket: `#bind_tcp`, `#bind_unused_port`
+# * TCP socket with TLS/SSL: `#bind_tls`
+# * Unix socket `#bind_unix`
+#
+# `#bind(uri : URI)` and `bind(uri : String)` parse socket configuration for
```
Comment (r00ster91) [inline src/http/server.cr]:
```suggestion
# `#bind(uri : URI)` and `#bind(uri : String)` parse socket configuration for
```

**Comment by straight-shoota** [comment]
Thanks @r00ster91 for prove reading! ❤️

File: src/http/server.cr [hunk 4]
```diff
@@ -54,14 +51,45 @@ require "./common"
...
+# ### Handler chain
+#
+# The handler given to a server can simply be a block that receives an `HTTP::Server::Context`,
+# or it can be an instance of `HTTP::Handler`. An `HTTP::Handler` has an optional
+# `#next` method to forward processing to the next handler in the chain.
+#
+# For example, an initial handler might handle exceptions raised from subsequent
+# handlers and return a `500 Server Error` status (see `HTTP::ErrorHandler`).
+# The next handler might log all incoming requests (see `HTTP::LogHandler`).
+# And the final handler deals with routing and application logic.
```
Comment (bcardiff) [inline src/http/server.cr]: We are dropping how the optional block interact with the rest of the handlers.

Comment (straight-shoota) [reply src/http/server.cr]: I don't follow. Are you referencing an intended change to the behaviour?

**Comment by RX14** [comment]
I think I'd prefer the ordering to talk about how the block form works first, and the role of `HTTP::Context` and then digress into `HTTP::Handler`, followed by talking about binding. I think this order reflects the order which a newbie would want to explore the server: first how to integrate your logic, since that's the interesting part. Then only after how to actually use the interface.

**Pull request merged**